### PR TITLE
v2

### LINKED ISSUE #32: Option to skip
While I get the value of this addon for new users, it's something every developer has to remove every time they spin up a new project. This has added an extra manual step that wasn't there before, and after the welcome page is seen once or twice, there's a diminishing value for subsequent appearances.

maybe `ember new --skip-welcome` or something, so we can have our cake and eat it too? An environment variable that "bypasses" the welcome page?

### LINKED ISSUE #57: Using welcome-page to teach more
Here is are some ideas for ember-welcome-page generated from an ember core team meeting today. We discussed simplifying the architecture so that:

 - the application blueprint would include an `application.hbs` file (because everyone needs to make one anyway, and providing a file ready-to-go is friendlier and less error prone).
 - the `application.hbs` blueprint would include an actual component that renders the welcome message, like:

    ```hbs
{{!-- The welcome-page component displays Ember's default welcome message. You should feel free to delete it. --}}
{{welcome-page}}
{{outlet}}
```
    The welcome message itself would invite the user to open `application.hbs` and make edits and see their results live reloaded in the browser. The benefit of this pattern is that users can immediately dive into the edit/test loop by poking at the template and seeing their changes taking effect.
 - This addon becomes simpler (no more middleware or shouldOverride checking). It would be a normal component-exporting addon, with one addition: it could use the `treeFor` APIs in `index.js` to only emit its code & assets in development builds (so that users who leave it in package.json don't end up with any unnecessary bytes).
 - future capabilities that have been discussed (like a built-in documentation browser) could be separate addons, we don't need to support those cases here.

### PR DESCRIPTION
**Pull request by locks**
Fixes #32.
Fixes #57.

- [x] Move to new component-based architecture
- [x] Update dependencies
- [ ] Squash commits
- [ ] Open follow up PR with version bump

screen shot 2016-10-21 at 10 55 08

### CODE REVIEW

File: addon/templates/components/welcome-page.hbs [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    <img src="/ember-welcome-page/images/construction.png" alt="Under construction">
+  </div>
+  <div class="welcome">
+    <h2 id="title">Congratulations, you made it!</h2>
+
+    <p>You&rsquo;ve officially spun up your very first Ember app :-)</p>
+    <p>You&rsquo;ve got one more decision to make: what do you want to do next? We&rsquo;d suggest one of the following to help you get going:</p>
+    <ol>
+      <li><a href="https://guides.emberjs.com/v<%= version %>/getting-started/quick-start/">Quick Start</a> - a quick introduction to how Ember works. Learn about defining your first route, writing a UI component and deploying your application.</li>
+      <li><a href="http://guides.emberjs.com/v<%= version %>/tutorial/ember-cli/">Ember Guides</a> - this is our more thorough, hands-on intro to Ember. Your crash course in Ember philosophy, background and some in-depth discussion of how things work (and why they work the way they do).</li>
```
Comment (acorncom) [inline addon/templates/components/welcome-page.hbs]: Nice work here so far. As a reminder, we have to grab this version as well ...

File: tests/dummy/app/templates/application.hbs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{{!-- The welcome-page component displays Ember's default welcome message. You should feel free to delete it. --}}
```
Comment (acorncom) [inline tests/dummy/app/templates/application.hbs]: Drop the "you should" prior to feel free ...

**Comment by acorncom** [comment]
Any idea re: why the tests aren't passing?  I'm 👍 on merging this at this point, but we probably need the application blueprint in for Ember-CLI as well ...

**Comment by locks** [comment]
@acorncom dunno, I'm still working on "features". this is super not ready for merging!

**Comment by locks** [comment]
@acorncom @ef4

File: index.js [hunk 3]
```diff
@@ -1,112 +1,23 @@
...
   },
 
-  /* requested tests (need to be watching in real-time preferably ...)
-   - Has an application.hbs been created?
-   - Has an index.hbs been created?
-   - Have any routes been created?
-   */
-  shouldOverride: function(jsExtensions, templateExtensions, folderPath) {
+  treeForAddon: function() {
+    if (EmberAddon.env() === 'production') { return false; }
```
Comment (ef4) [inline index.js]: We probably want to do this same thing for the vendor tree too.

File: index.js [hunk 4]
```diff
@@ -1,112 +1,23 @@
...
-
-    // we'll assume we're running '/' but then bail if we aren't supposed to be overriding
-    app.get(options.baseURL, function(req, res, next) {
-      if (_this.shouldOverride(_this.jsExt, _this.templateExt, _this.app.project.root)) {
-        res.send(welcomePageHTML);
-      } else {
-        next();
-      }
-    });
+    app.import('vendor/construction.png');
```
Comment (ef4) [inline index.js]: I think this can happen only in non-production builds.

File: index.js (hunk 3 above)
Comment (locks) [inline index.js]: from my manual tests I believe the above also removes the vendor paths, hence me not guarding `app.import`in the `included` hook.

Comment (ef4) [inline index.js]: I thought `app.import` will cause an exception if the file is missing?

File: README.md [hunk 5]
```diff
@@ -1,23 +1,28 @@
 # Ember Welcome Page
 [![Build Status](https://travis-ci.org/ember-cli/ember-welcome-page.svg?branch=master)](https://travis-ci.org/ember-cli/ember-welcome-page)
 
-Displays a nice welcome page for newly generated applications with links to resources.
+This README outlines the details of collaborating on this Ember addon.
```
Comment (acorncom) [inline README.md]: Can we go back to what we had here?

File: README.md [hunk 6]
```diff
@@ -1,23 +1,28 @@
...
 
 ## Installation
 
-* `git clone` this repository
+* `git clone <repository-url>` this repository
+* `cd ember-welcome-page`
 * `npm install`
 * `bower install`
```
Comment (acorncom) [inline README.md]: Don't think we'd need bower?

File: addon/templates/components/welcome-page.hbs [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    .tomster img {
+      width: 60%;
+    }
+    .welcome, .tomster {
+      width: 100%;
+      float: none;
+      margin: auto;
+    }
+  }
+</style>
```
Comment (acorncom) [inline addon/templates/components/welcome-page.hbs]: Does this use of style in the body work across browsers? It's non standard I believe ...

File: addon/templates/components/welcome-page.hbs [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+      <li><a href="http://guides.emberjs.com/v{{emberVersion}}/tutorial/ember-cli/">Ember Guides</a> - this is our more thorough, hands-on intro to Ember. Your crash course in Ember philosophy, background and some in-depth discussion of how things work (and why they work the way they do).</li>
+    </ol>
+    <p>If you run into problems, you can check <a href="http://stackoverflow.com/questions/tagged/ember.js">Stack Overflow</a> or <a href="http://discuss.emberjs.com/">our forums</a>  for ideas and answers—someone&rsquo;s probably been through the same thing and already posted an answer.  If not, you can post your <strong>own</strong> question. People love to help new Ember developers get started, and our community is incredibly supportive <3</p>
+  </div>
+  <p class="postscript">To remove this welcome message, remove the <code>\{{welcome-page}}</code> component from you <code>application.hbs</code> file.<br>You'll see this page update soon after!</p>
```
Comment (acorncom) [inline addon/templates/components/welcome-page.hbs]: from **your

File: index.js [hunk 9]
```diff
@@ -1,112 +1,29 @@
...
-
-    var routerPath = path.join(folderPath, 'app', 'router.js');
-    var hasRouterMapContents = false;
-
-    if (existsSync(routerPath)) {
-      var router = fs.readFileSync(routerPath, { encoding: 'utf8' });
-      hasRouterMapContents = router.match(/this\.route|this\.resource/);
-    }
+  treeForAddon: function() {
+    if (EmberAddon.env() === 'production') { return false; }
```
Comment (acorncom) [inline index.js]: Do you want to use `_isProduction` here?

File: README.md (hunk 6 above)
Comment (acorncom) [inline README.md]: Never mind on this

**Comment by acorncom** [comment]
@locks one spelling mistake here that we *should* fix, the rest are nitpicks that have no bearing on shipping. Nice work

**Comment by acorncom** [comment]
LGTM

**Pull request merged**